acceptance/tests/resource/package/common_package_name_in_different_providers.rb

```diff
@@ -0,0 +1,107 @@
+test_name "ticket 1073: common package name in two different providers should be allowed"
+
+hosts_to_test = agents.select do |agent|
+  agent['platform'].match /(?:centos|el-|fedora)/
+end
+skip_test "No suitable hosts found" if hosts_to_test.empty?
+
+require 'puppet/acceptance/rpm_util'
+extend Puppet::Acceptance::RpmUtils
+
+rpm_options = {:pkg => 'guid', :version => '1.0'}
+
+teardown do
+  step "cleanup"
+  agents.each do |agent|
+    clean_rpm agent, rpm_options
+  end
+end
+
+def verify_state(hosts, pkg, state, match)
+  hosts.each do |agent|
+    on agent, puppet('resource', 'package', pkg, '--param provider') do
+      assert_match(%r@package\s*\{\s*'#{pkg}':\s*ensure\s*=>\s*'#{state}',\s*provider\s*=>\s*'yum',\s*\}@m, stdout)
```

Comment:
That sounds fine, I'll give it a try. Did we also try to verify that
resource returned the right information in the ips tests?

On Fri, Jan 2, 2015 at 10:57 AM, Kylo Ginsberg <notifications@github.com>
wrote:

> In
> acceptance/tests/resource/package/common_package_name_in_different_providers.rb
> <https://github.com/puppetlabs/puppet/pull/3353#discussion-diff-22418660>:
>
> > +require 'puppet/acceptance/rpm_util'
> > +extend Puppet::Acceptance::RpmUtils
> > +
> > +rpm_options = {:pkg => 'guid', :version => '1.0'}
> > +
> > +teardown do
> > +  step "cleanup"
> > +  agents.each do |agent|
> > +    clean_rpm agent, rpm_options
> > +  end
> > +end
> > +
> > +def verify_state(hosts, pkg, state, match)
> > +  hosts.each do |agent|
> > +    on agent, puppet('resource', 'package', pkg, '--param provider') do
> > +      assert_match(%r@package\s*\{\s*'#{pkg}':\s*ensure\s*=>\s*'#{state}',\s*provider\s*=>\s*'yum',\s*\}@m, stdout)
>
> What do you think about using yum itself in verify_state rather than
> puppet? That way we're not using puppet to verify puppet. This is the model
> that the acceptance/tests/resource/package/ips used, where we try
> something via puppet, then verify it via the underlying tools.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3353/files#r22418660>.
>